x-pack/plugins/observability_solution/entity_manager/server/routes/enablement/disable.ts

```diff
@@ -58,10 +50,14 @@ export function disableEntityDiscoveryRoute<T extends RequestHandlerContext>({
           deleteData: req.query.deleteData,
         });
 
-        await deleteEntityDiscoveryAPIKey((await context.core).savedObjects.client);
-        await server.security.authc.apiKeys.invalidateAsInternalUser({
-          ids: [apiKey.id],
-        });
+        server.logger.debug('reading entity discovery API key from saved object');
+        const apiKey = await readEntityDiscoveryAPIKey(server);
+        if (apiKey) {
+          await deleteEntityDiscoveryAPIKey(soClient);
+          await server.security.authc.apiKeys.invalidate(req, {
```

Comment:
can invalidate with current user since we've verified they had necessary permissions